chia/full_node/full_node_api.py

```diff
@@ -321,12 +326,19 @@ async def request_blocks(self, request: full_node_protocol.RequestBlocks) -> Opt
                     reject = RejectBlocks(request.start_height, request.end_height)
                     msg = make_msg(ProtocolMessageTypes.reject_blocks, reject)
                     return msg
-                blocks.append(block_bytes)
 
-        msg = make_msg(
-            ProtocolMessageTypes.respond_blocks,
-            full_node_protocol.RespondBlocks(request.start_height, request.end_height, blocks),
-        )
+                self.log.warning(f"Type: {type(block_bytes)}")
+                blocks_bytes.append(block_bytes)
+
+            respond_blocks_manually_streamed: bytes = (
+                bytes(uint32(request.start_height))
+                + bytes(uint32(request.end_height))
+                + len(blocks_bytes).to_bytes(4, "big", signed=False)
+            )
+            for block_bytes in blocks_bytes:
+                respond_blocks_manually_streamed += block_bytes
```

Comment:
does manually streaming make a difference here? it doesn't look like the blocks themselves were deserialized + serialized previously at least.

Reply:
Yeah they were. If you initialize a streamable RespondBlocks(height, height, blocks) where blocks is a list of bytes, it would recursively parse every single FullBlock from bytes to a FullBlock object. This is a feature of streamable (in post_init), where if you pass in bytes, it constructs the right object. 

Here we are avoiding ever initializing a FullBlock object.